### ISSUE TITLE
Webpack loader for Angular2 AOT compilation?

### ISSUE THREAD

**Issue by IAMtheIAM** [webpack/webpack#2918]
**I'm submitting a feature request**

**Webpack version:**
1.10.x/2.x

**Please tell us about your environment:**
OSX 10.x / Linux / Windows 10

**Current behavior:**
Webpack does not work with Angular2 Ahead of Time compilation

**Expected/desired behavior:**
The latest release of Angular2 allows for Ahead of time (AOT) compilation. Can we get this feature to integrate with webpack and typescript loaders? What do we need to do
- **What is the motivation / use case for changing the behavior?**
  Increase production app loading speed, decrease js bundle size, use latest features.
- **Browser:** [all ] 
- **Language:** [all | TypeScript X.X | ES6/7 | ES5 | Dart | ...]

**Comment by TheLarkInn** [comment]
Unfortunately despite getting a 3 month warning from me personally about this, angular AoT compiler does not have endpoints open for it to be implemented the correct way through webpack loaders the _right way_. I'll reference this issue: https://github.com/angular/angular/issues/8759

**Comment by blacksonic** [comment]
It works now :) Check out the @ngtools/webpack peckage in [angular-cli](https://github.com/angular/angular-cli) or my [HelloWorld example](https://github.com/blacksonic/angular2-aot-webpack) with it.

**Comment by BrainCrumbz** [comment]
Has anyone an example of a Webpack 2 *production* build (i.e. with UglifyJsPlugin), with Angular2 AOT enabled and a *global CSS* (not one from a component) included with `require('...css');` ? In our team we cannot seem to find an example of that. We got a working non-AOT build, but adding AOT to the equation breaks it (*'require' name cannot be found*).

**Comment by tinchou** [comment]
You can have a loader for that specific file using include. In my case I
use extract-text-plugin to combine multiple CSS files into one output file.

Then I just include that file from index.html, but don’t use require since
it doesn’t work with AoT.
​

**Comment by BrainCrumbz** [comment]
Thanks for suggestion. I did not yet wrap my head around extract-text-plugin, to say the truth, and how it combines with all other loaders. So I guess will have to read about that deeper.

In meantime, we were able to overcome the mentioned issue by switching from `require(...css);` to `import ...css;`. Doing so there was no require problem, obviously, and every step in build chain seems to be happy, both in dev build (without AOT nor uglify) and in production build (with AOT and uglify). Opening web page shows that global CSS is picked up. 

Now running into another a by-product issue: with the following loaders for global CSSes and with source-map enabled: 

```
    loaders: [
      'style-loader',
      'css-loader?sourceMap',
      'postcss-loader?sourceMap',
    ],
```

css-loader complains with `Module build failed: TypeError: Path must be a string. Received undefined`. But I think that belongs to another place.
